File: Problem 7/Python/problem7.py
#A function to test the primality of a given number...
def isPrime(k):
	factors = 0
	#We only really care about [1,sqrt(k)+1]
	for i in range(1, int(k**.5 + 1)):
		#i is a factor of k
		if k % i == 0:
			factors+=1
		#We aren't prime
		if factors >= 2:
			return False
	#Prime!
	return True

def main():
	numPrimes = 1
	prime = 2
	num = 3

	while numPrimes < 10001:
		if isPrime(num):
			numPrimes+=1
			prime = num
		num+=2
	print(prime)

if __name__ == '__main__':
	main()

File: Problem 7/Nim/problem7.nim
import math

#Determine if a given number is prime
proc isPrime(n: int64): bool =
    var factors: int = 0
    if n mod 2 == 0 and n != 2:
        return false
    for i in 1..int (sqrt(float n) + 1):
        if n mod i == 0:
            factors += 1
    return factors < 2

#Find the nth prime
proc problem7(n: int): int64 = 
    var index = 2
    var currentPrime = 3
    var currentNumber = 4
    while index < n:
        if isPrime(currentNumber):
            currentPrime = currentNumber
            index += 1
        currentNumber += 1
    return currentPrime

echo(problem7(10_001))